interfaces/input_parameters/ring.py

```diff
@@ -182,182 +130,348 @@ class Ring:
+
+                # Setting the Ring as attribute of the sections for reference
+                section._Ring = self
+
+        self.Section_list = Section_list
+        self.n_sections = len(self.Section_list)
+
+        # Extracting the length of sections to get circumference
+        self.section_length_design = np.array(
+            [section.length_design for section in self.Section_list])
+        self.circumference = np.sum(self.section_length_design)
+
+        # Computing ring radius
+        self.radius = self.circumference / (2 * np.pi)
+
+        # Extracting the bending radius from all sections
+        self.bending_radius = np.array([
+            section.bending_radius for section in self.Section_list])
+
+        # Extracting the synchronous data from the sections
+        for index_section, section in enumerate(self.Section_list):
+            if index_section == 0:
+                self.synchronous_data = section.synchronous_data
+            else:
+                if (section.synchronous_data == self.synchronous_data).all:
+                    pass
+                else:
+                    warn_message = 'The synchronous data for all sections ' + \
+                        'is not identical. This case is not yet fully ' + \
+                        'tested and is under implementation.'
+                    warnings.warn(warn_message)
+
+        # Processing the momentum program
+        # Getting the options to get at which time samples the interpolation
+        # is made
         t_start = kwargs.pop('t_start', 0)
         t_stop = kwargs.pop('t_stop', np.inf)
         interp_time = kwargs.pop('interp_time', 0)
 
+        # Treating the input of the interp_time option
         if not hasattr(interp_time, '__iter__'):
             interp_time = (interp_time, )
 
+        # Getting the sample times
         sample_func, start, stop = tmng.time_from_sampling(*interp_time)
 
+        # Keeping only the values between the user defined bounds
         if t_start > start:
             start = t_start
         if t_stop < stop:
             stop = t_stop
 
-        synchronous_data.convert(self.Particle.mass, self.Particle.charge,
-                                 self.bending_radius)
-
-        interpolation = kwargs.pop("interpolation", 'linear')
+        # Getting options for the interpolation
+        interpolation = kwargs.pop('interpolation', 'linear')
         store_turns = kwargs.pop('store_turns', True)
 
-        self.momentum = synchronous_data.preprocess(
+        # Processing the momentum program and interpolating on the
+        # values defined by sample_func
+        momentum_processed = self.synchronous_data.preprocess(
             self.Particle.mass,
-            self.ring_circumference, sample_func,
+            self.circumference, sample_func,
             interpolation, start, stop,
             store_turns=store_turns)
 
-        self.n_sections = self.momentum.shape[0]-2
```

Comment:
I reckon it's worth adding a sanity check for the sections, something like:
```python
if self.n_sections != momentum_processed.shape[0]-2:
    raise except.BuggeredInterpolationError("message")
```

Reply:
Indeed, I was thinking of a more general routine in the end of the __init__ to check the consistency (e.g. that the shape of all important arrays after processing is ok).

To be consolidated on the functioning code